packages/ui/src/common/legal-accepted.tsx

```diff
@@ -0,0 +1,77 @@
+import * as Common from '@clerk/elements/common';
+import { cx } from 'cva';
+import React from 'react';
+
+import { useAppearance } from '~/contexts';
+import { useEnvironment } from '~/hooks/use-environment';
+import { useLocalizations } from '~/hooks/use-localizations';
+import * as Field from '~/primitives/field';
+
+import { LinkRenderer } from './link-renderer';
+
+export function LegalAcceptedField({
+  className,
+  checked = false,
+  ...restProps
+}: Omit<React.ComponentProps<typeof Common.Input>, 'type'>) {
+  const { t } = useLocalizations();
+  const { displayConfig } = useEnvironment();
+  const { parsedAppearance } = useAppearance();
+  const termsUrl = parsedAppearance.options.termsPageUrl || displayConfig.termsUrl;
+  const privacyPolicyUrl = parsedAppearance.options.privacyPageUrl || displayConfig.privacyPolicyUrl;
+
+  let localizedText: string | undefined;
+
+  if (termsUrl && privacyPolicyUrl) {
+    localizedText = t('signUp.__experimental_legalConsent.checkbox.label__termsOfServiceAndPrivacyPolicy', {
+      termsOfServiceLink: termsUrl,
+      privacyPolicyLink: privacyPolicyUrl,
+    });
+  } else if (termsUrl) {
+    localizedText = t('signUp.__experimental_legalConsent.checkbox.label__onlyTermsOfService', {
+      termsOfServiceLink: termsUrl,
+    });
+  } else if (privacyPolicyUrl) {
+    localizedText = t('signUp.__experimental_legalConsent.checkbox.label__onlyPrivacyPolicy', {
+      privacyPolicyLink: privacyPolicyUrl,
+    });
+  }
+
+  return (
+    <Common.Field
+      name='__experimental_legalAccepted'
+      asChild
+    >
+      <Field.Root>
+        <div className='flex justify-center gap-2'>
```

Comment:
Will want to replace the `gap-2` usage here with padding left on the label.

Reply:
Makes sense!